test/unittests/iotools/test_bids_utils.py

```diff
@@ -38,6 +38,47 @@
 )
 
 
+def test_create_scans_dict_error(tmp_path):
+    from clinica.iotools.bids_utils import create_scans_dict
+
+    with pytest.raises(
+        ValueError,
+        match=f"Dataset foo is not supported.",
+    ):
+        create_scans_dict(str(tmp_path), "foo", str(tmp_path), [], "", "", {})
```

Comment:
Improve test flexibility on dataset name.
```suggestion
    dataset = "foo"

    with pytest.raises(
        ValueError,
        match=f"Dataset {dataset} is not supported.",
    ):
        create_scans_dict(str(tmp_path), dataset, str(tmp_path), [], "", "", {})
```